psi/pmt_test.go

```diff
@@ -520,6 +564,45 @@ func TestFilterPMTPacketsToPids_MultiPacketPMT(t *testing.T) {
 	}
 }
 
+func TestFilterPMTPacketsToPids_PIDNotFound(t *testing.T) {
+	// PMT contains PIDs 101-105.
+	pmtPacketBytes, _ := hex.DecodeString("4740641D0002B0940001DF0000E065F0050E03C015581BE065F0150E03C0109D2A027E1F9700E9080C001F418503E84187E066F01A0E03C00122050445414333CC07E0C2B0E8656E670A04656E670087E067F01A0E03C00122050445414333CC07E0C2B0E8656E670A04656E67000FE068F0100E03C001262B030102010A04656E67000FE069F0100E03C001262B030102010A04656E67002E9B5B71FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")
+
+	// Copy into a packet.Packet for calling FilterPMTPacketsToPids.
+	var pkt packet.Packet
+	copy(pkt[:], pmtPacketBytes)
+
+	acc := packet.NewAccumulator(PmtAccumulatorDoneFunc)
+	done, err := acc.Add(pmtPacketBytes)
+
+	if err != nil {
+		t.Error(err)
+	}
+	if !done {
+		t.Errorf("Single packet PMT expected. This means your doneFunc is probably bad.")
+	}
+
+	// Test when some of the PIDs to be filtered exist in the PMT.
+	pmtPkts, err := FilterPMTPacketsToPids([]*packet.Packet{&pkt}, []uint16{105, 106})
+	if err.Error() != "PID(s) [106] not found in PMT." {
+		t.Errorf("Expected missing PID error string, got %s", err.Error())
+	}
+
+	if pmtPkts == nil {
+		t.Error("Expected non-nil packets since none of the filtered PIDs exist in the PMT.")
```

Comment:
should the error message here be? Expected non-nil packets since _some_ of the filtered PIDs exist in the PMT.

Reply:
Indeed it should.  Good catch.